app/controllers/books/book/pages/new.js

```diff
@@ -0,0 +1,22 @@
+import Controller from '@ember/controller';
+import { inject as service } from '@ember/service';
+
+export default Controller.extend({
+
+  session: service(),
+  routing: service('-routing'),
+
+  actions: {
+    createPage() {
+
+      let newPage = this.store.createRecord('page', {
```

Comment:
If it does not bring a lot of extra readability, try to avoid memoizing just for sake of memoizing (it is not repeated anywhere, so you could run `save` directly after `createRecord`.